File: exercise/python/day22/eid/sex.py
from enum import Enum
from typing import Union

from eid.parsing_error import ParsingError


class Sex(Enum):
    Sloubi = 1
    Gagna = 2
    Catact = 3


def parse_sex(potential_sex: str) -> Union[ParsingError, Sex]:
    if potential_sex == '1':
        return Sex.Sloubi
    elif potential_sex == '2':
        return Sex.Gagna
    elif potential_sex == '3':
        return Sex.Catact
    else:
        return ParsingError("Not a valid sex")

File: exercise/ts/day22/src/eid/sex.ts
import {Either, left, right} from 'fp-ts/Either';
import {ParsingException} from "./parsingException";

export enum Sex {
    Sloubi = '1',
    Gagna = '2',
    Catact = '3'
}

export namespace Sex {
    export function parse(potentialSex: string): Either<ParsingException, Sex> {
        switch (potentialSex) {
            case '1':
                return right(Sex.Sloubi);
            case '2':
                return right(Sex.Gagna);
            case '3':
                return right(Sex.Catact);
            default:
                return left(new ParsingException('Not a valid sex'));
        }
    }
}